frontend/src/metabase/browse/components/RecentlyViewedModels.tsx

```diff
@@ -0,0 +1,75 @@
+import { skipToken, useListRecentItemsQuery } from "metabase/api";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import { Box, Text } from "metabase/ui";
+
+import type { ModelResult, RecentModel } from "../types";
+import { getIcon } from "../utils";
+
+import { ModelCard } from "./ModelCard";
+import { RecentlyViewedModelsGrid } from "./RecentlyViewedModels.styled";
+import { getCountOfRecentlyViewedModelsToShow } from "./utils";
+
+export const RecentlyViewedModels = ({
+  modelCount,
+  applyCurrentFilters,
+}: {
+  /** The number of recently viewed models shown
+   * depends on the number of models shown in the table below. */
+  modelCount: number;
+  applyCurrentFilters: (unfilteredModels: ModelResult[]) => ModelResult[];
```

Comment:
I think this gets simpler on a few levels if you do the data fetching and filtering one component up, and simply pass the already filtered models into this (now) pure component.

Reply:
I began to do this but I ran into a complication. By moving state up, we make it a bit awkward to give the RecentlyViewedModels component its own error state.

Do you think this refactor is a blocker?